File: src/bot/commands/seek.ts
import type { Message } from 'discord.js-selfbot-v13';
import { getVideoStreamer } from '../../stream/video-streamer.js';
import { parseTimeString, formatDuration } from '../../plex/library.js';

export async function seekCommand(message: Message, args: string[]): Promise<void> {
  if (!message.guild) {
    await message.channel.send('❌ This command can only be used in a server');
    return;
  }

  const timeArg = args[0];
  if (!timeArg) {
    await message.channel.send('❌ Usage: `!seek <time>` (e.g., `!seek 1:30:00` or `!seek 45:00`)');
    return;
  }

  const guildId = message.guild.id;
  const videoStreamer = getVideoStreamer();
  const session = videoStreamer.getSession(guildId);

  if (!session) {
    await message.channel.send('❌ Nothing is currently playing');
    return;
  }

  const timeMs = parseTimeString(timeArg);

  if (timeMs === null) {
    await message.channel.send('❌ Invalid time format. Use `HH:MM:SS`, `MM:SS`, or seconds');
    return;
  }

  const statusMsg = await message.channel.send(`⏩ Seeking to ${formatDuration(timeMs)}...`);

  const success = await videoStreamer.seekStream(guildId, timeMs);

  if (success) {
    await statusMsg.edit(`⏩ Seeked to ${formatDuration(timeMs)}`);
  } else {
    await statusMsg.edit('❌ Failed to seek');
  }
}

File: selfbot/commands/seek.py
"""Seek command via IPC."""

from __future__ import annotations

from typing import TYPE_CHECKING
import discord

from ..ipc import ipc
from ..helpers import parse_time_string, format_duration

if TYPE_CHECKING:
    from ..client import SchroStreamBot


async def seek_command(bot: SchroStreamBot, message: discord.Message, args: list[str]) -> None:
    if not message.guild:
        await message.channel.send("❌ This command can only be used in a server")
        return

    if not args:
        await message.channel.send("❌ Usage: `!seek <time>` (e.g., `!seek 1:30:00` or `!seek 45:00`)")
        return

    time_ms = parse_time_string(args[0])
    if time_ms is None:
        await message.channel.send("❌ Invalid time format. Use `HH:MM:SS`, `MM:SS`, or seconds")
        return

    status_msg = await message.channel.send(f"⏩ Seeking to {format_duration(time_ms)}...")

    result = await ipc.control_seek(str(message.guild.id), time_ms)

    if result.get("success"):
        await status_msg.edit(content=f"⏩ Seeked to {format_duration(time_ms)}")
    else:
        await status_msg.edit(content="❌ Failed to seek")
